vsutil/__init__.py

```diff
@@ -36,12 +38,38 @@ class Dither(Enum):
     ERROR_DIFFUSION = 'error_diffusion'  # Floyd-Steinberg error diffusion.
 
 
+def disallow_variable_format(function: Callable[..., R]) -> Callable[..., R]:
```

Comment:
For proper type preservation, you should use a typevar for `Callable[..., R]`. See https://mypy.readthedocs.io/en/stable/generics.html#declaring-decorators.

Reply:
https://github.com/Irrational-Encoding-Wizardry/vsutil/pull/37/commits/93324264cd82cb6afaeee010657747b6f2e64862 look good?